doc/Manuals/LoadBalancing/ig-xlb_x-road_external_load_balancer_installation_guide.md

```diff
@@ -0,0 +1,773 @@
+health check will start returning failures. This is a feature of caching and not a bug per se. In addition, logging out
+of a security server's keys should not occur by accident so it should not be a surprise that the node cannot send messages
+after not having access to it's keys.
+
+
+#### 3.4.2 Health check examples
+
+Before testing with an actual load balancer, you can test the health check service with `curl`, for example.
+
+Health check service response when everything is up and running and messages should go through.
+```
+$ curl -i localhost:5588
+   HTTP/1.1 200 OK
+   Content-Length: 0
+   Server: Jetty(8.y.z-SNAPSHOT)
+```
+
+Health check service response when the service `xroad-signer` is not running.
+```
+$ curl -i localhost:5588
+HTTP/1.1 500 Server Error
+Transfer-Encoding: chunked
+Server: Jetty(8.y.z-SNAPSHOT)
+
+Fetching health check response timed out for: Authentication key OCSP status
+```
+
+
+## 4. Database replication setup
+
+For technical details on the PostgreSQL replication, refer to the [official documentation](https://www.postgresql.org/docs/9.2/static/high-availability.html).
+Note that the versions of PostgreSQL distributed with RHEL and Ubuntu are different. At the time of writing, RHEL 7
+distributes PostgreSQL version 9.2 and Ubuntu 14.04 distributes version 9.3; the replication configuration is the same
+for both versions.
+
+
+### 4.1 Setting up TLS certificates for database authentication
+This section describes how to create and set up certificate authentication between the slave and master database instances.
+
+For further details on the certificate authentication, see the
+[PostgreSQL documentation](https://www.postgresql.org/docs/9.2/static/auth-methods.html#AUTH-CERT).
+
+
+1. Generate the Certificate Authority key and a self-signed certificate for the root-of-trust:
+
+   ```
+   openssl req -new -x509 -days 7300 -nodes -sha256 -out master.crt -keyout master.key -subj '/O=cluster/CN=master'
+   ```
+   The subject name does not really matter here. Remember to keep the `master.key` file in a safe place.
+
+2. Generate keys and certificates signed by the CA for each postgresql instance, including the master. Do not use the CA
+   certificate and key as the database certificate and key.
+
+   ```
+   openssl req -new -nodes -days 7300 -keyout server.key -out server.csr -subj "/O=cluster/CN=<nodename>"
+   ```
+
+   **Note:** The `<nodename>` must match a replication user name; otherwise the subject name does not matter
+
+   ```
+   openssl x509 -req -in server.csr -CAcreateserial -CA master.crt -CAkey master.key -days 7300 -out server.crt
+   ```
+
+3. Copy the certificates to `/etc/xroad/postgresql` on each cluster instance:
+
+   ```bash
+   sudo mkdir -p -m 0755 /etc/xroad/postgresql
+   sudo chmod o+x /etc/xroad
+   ```
+   Copy the certificates, then copy one server key per instance so that each instance has a unique key:
```

Comment:
Maybe explicitly mention _which certificates_ should be copied to /etc/xroad/postgresql? Seems I only copied server.crt, but left out master.crt, which I found out to be wrong later.